src/RawPOSIX/src/safeposix/syscalls/net_calls.rs

```diff
@@ -135,7 +138,7 @@ impl Cage {
      *   Get the kernel fd with provided virtual fd first
      *   connect() will return 0 when success and -1 when fail
      */
-    pub fn connect_syscall(&self, virtual_fd: i32, addr: &GenSockaddr) -> i32 {
+    pub fn connect_syscall(&self, virtual_fd: i32, addr: &GenSockaddr, timeout_ms: Option<i32>) -> i32 {
```

Comment:
With all of these we definitely don't want to add a new parameter to the function signature.

We also dont need to write our own timeout mechanisms since they are provided via setsockopt. Check here to see how that's done https://github.com/Lind-Project/safeposix-rust/blob/cd70d39a364bec626433ccf2f5b99788bb68bbdf/src/interface/comm.rs#L236

Basically you can just set this whenever a socket is created and then just loop to check if it times out within the syscalls.